### PR TITLE
Resource Deployment UX Refresh

### PR DESCRIPTION
**Pull request by aasimkhan30**
First pass at the resource deployment UX refresh. This PR changes the card style to that of azure portal ones. 

Screenshots:

Light not selected: 
image
Light selected:
image
Dark not selected:
image
dark selected
image

Resource dialog light
image
Resource dialog dark
image

### CODE REVIEW

File: src/sql/azdata.proposed.d.ts [hunk 1]
```diff
@@ -261,7 +261,8 @@ declare module 'azdata' {
 		iconWidth?: string;
 		iconHeight?: string;
 		selectedCardId?: string;
-		orientation?: Orientation // Defaults to horizontal
+		orientation?: Orientation; // Defaults to horizontal
+		cardStyle?: 'regular' | 'portalCard'; // Defaults to regular
```
Comment (alanrenmsft) [inline src/sql/azdata.proposed.d.ts:265]: I'd prefer to call it regular/detail

Comment (alanrenmsft) [inline src/sql/azdata.proposed.d.ts:265]: I am not a fan of the name portalcard. the difference between the old and new is the icon position, maybe we can call this property: IconPosition: top|left?

Comment (aasimkhan30) [reply src/sql/azdata.proposed.d.ts:265]: Yeah, sounds good to me. I couldn't think of a better while I was coding.

Comment (alanrenmsft) [reply src/sql/azdata.proposed.d.ts:265]: cool, and don't forgot to do a bulk find/replace of the css class names

Comment (ninarn) [reply src/sql/azdata.proposed.d.ts:265]: I agree with not calling it a portal card :)

File: src/sql/workbench/browser/modelComponents/media/card.css [hunk 2]
```diff
@@ -221,3 +221,38 @@
 	text-align: right;
 	white-space: nowrap;
 }
+
+
+.model-left-icon-card {
```
Comment (aaomidi) [inline src/sql/workbench/browser/modelComponents/media/card.css:226]: please add .horizontal before all of these, the styles are not necessarily translatable between horizontal and vertical. See above in the CSS file for examples

File: src/sql/workbench/browser/modelComponents/radioCardGroup.component.ts [hunk 3]
```diff
@@ -119,6 +119,18 @@ export default class RadioCardGroup extends ComponentBase implements IComponent,
 		return this.getSpecficProperties().selectedCardId ?? undefined;
 	}
 
+	public get iconPosition(): string {
+		return this.getPropertyOrDefault<azdata.RadioCardGroupComponentProperties, string>((props) => props.iconPosition, 'top');
```
Comment (aaomidi) [inline src/sql/workbench/browser/modelComponents/radioCardGroup.component.ts:123]: Just use getSpecificProperties()

File: src/sql/azdata.proposed.d.ts [hunk 4]
```diff
@@ -261,7 +261,8 @@ declare module 'azdata' {
 		iconWidth?: string;
 		iconHeight?: string;
 		selectedCardId?: string;
-		orientation?: Orientation // Defaults to horizontal
+		orientation?: Orientation; // Defaults to horizontal
+		iconPosition?: 'top' | 'left'; // Defaults to top
```
Comment (aaomidi) [inline src/sql/azdata.proposed.d.ts:265]: How does this work with the orientation is vertical?

**Review by aaomidi — CHANGES_REQUESTED** [review]
This PR has broken the vertical layout right now:

image

**Review by ninarn — APPROVED** [review]

**Review by alanrenmsft — APPROVED** [review]

**Pull request merged**